src/libunwind.cpp

```diff
@@ -218,6 +218,27 @@ _LIBUNWIND_HIDDEN int __unw_resume(unw_cursor_t *cursor) {
 }
 _LIBUNWIND_WEAK_ALIAS(__unw_resume, unw_resume)
 
+/// ClickHouse: Convenient helper (added for jemalloc)
+int unw_backtrace(void **buffer, int size) {
```

Comment:
The question is - what prevents having this helper outside of this library in our code? Does it use private API?

Reply:
Doesn't use private API. IIUC:
 * this function is present in the original libunwind, but not in llvm-libunwind,
 * jemalloc uses it: `contrib/jemalloc/src/prof_sys.c` includes `<libunwind.h>` and calls unw_backtrace().

So we have to put it in `libunwind.h`, or patch jemalloc, or (if I'm reading the code correctly) use `mallctl("experimental.hooks.prof_backtrace", nullptr, nullptr, &unw_backtrace, 8)` to provide the function at runtime.

Reply:
>jemalloc uses it: contrib/jemalloc/src/prof_sys.c includes <libunwind.h> and calls unw_backtrace().

It does not uses unw_backtrace, since `JEMALLOC_PROF_LIBUNWIND` is not defined, `JEMALLOC_PROF_LIBGCC` is used - https://github.com/ClickHouse/ClickHouse/blob/9e9862066fcf5fae5342289c6134627b335126f2/contrib/jemalloc-cmake/CMakeLists.txt#L182-L187

Example of stacktrace - https://pastila.nl/?00d7b55b/044b1c710ec1b6f40a1142ba6d8555ab#1spY2b4S8szuWI9B3XSdRA== (by @antonio2368 )

Reply:
Thanks for pointing out! There are 2 more uses of unw_backtrace, both in our own code:
 * `contrib/llvm-project/libcxx/include/exception`
 * `src/Common/StackTrace.cpp`
After removing these it builds. We should probably replace them both with `_Unwind_Backtrace`.

Reply:
Egh, now, it is used - https://github.com/ClickHouse/ClickHouse/pull/66346

So I'm suggesting to leave it as-is for now, until all the callers will be fixed